src/components/Search/SearchRouter/SearchRouterList.tsx

```diff
@@ -0,0 +1,125 @@
+import type {ListItemWithSingleIcon, SingleIconListItemProps} from '@components/SelectionList/Search/SingleIconListItem';
+import type {SectionListDataType, UserListItemProps} from '@components/SelectionList/types';
+import UserListItem from '@components/SelectionList/UserListItem';
+import useLocalize from '@hooks/useLocalize';
+import useThemeStyles from '@hooks/useThemeStyles';
+import Navigation from '@libs/Navigation/Navigation';
+import type {OptionData} from '@libs/ReportUtils';
+import * as SearchUtils from '@libs/SearchUtils';
+import * as Report from '@userActions/Report';
+import ROUTES from '@src/ROUTES';
+
+type ItemWithQuery = {
+    query: string;
+};
+
+type SearchRouterListProps = {
+    currentSearch: SearchQueryJSON | undefined;
+    reportForContextualSearch?: OptionData;
+    recentSearches: ItemWithQuery[] | undefined;
+    recentReports: OptionData[];
+    onRecentSearchSelect: (query: SearchQueryJSON | undefined, shouldAddToRecentSearch?: boolean) => void;
+    closeAndClearRouter: () => void;
+};
+
+function SearchRouterItem(props: UserListItemProps<OptionData> | SingleIconListItemProps<ListItemWithSingleIcon & ItemWithQuery>) {
+    const styles = useThemeStyles();
+
+    if ('item' in props && props.item.reportID) {
+        return (
+            <UserListItem
+                pressableStyle={styles.br2}
+                // eslint-disable-next-line react/jsx-props-no-spreading
+                {...(props as UserListItemProps<OptionData>)}
+            />
+        );
+    }
+    // eslint-disable-next-line react/jsx-props-no-spreading
+    return <SingleIconListItem {...(props as SingleIconListItemProps<ListItemWithSingleIcon & ItemWithQuery>)} />;
+}
+
+function SearchRouterList({currentSearch, reportForContextualSearch, recentSearches, recentReports, onRecentSearchSelect, closeAndClearRouter}: SearchRouterListProps) {
+    const styles = useThemeStyles();
+    const {translate} = useLocalize();
+    const sections: Array<SectionListDataType<OptionData | (ListItemWithSingleIcon & ItemWithQuery)>> = [];
+
+    if (currentSearch?.inputQuery) {
+        sections.push({
+            data: [
+                {
+                    text: currentSearch?.inputQuery,
+                    singleIcon: Expensicons.MagnifyingGlass,
+                    query: currentSearch?.inputQuery,
+                    itemStyle: styles.activeComponentBG,
+                    keyForList: 'findItem',
+                },
+            ],
+        });
+    }
+
+    if (reportForContextualSearch) {
+        sections.push({
+            data: [
+                {
+                    text: `${translate('search.searchIn')}${reportForContextualSearch.text ?? reportForContextualSearch.alternateText}`,
+                    singleIcon: Expensicons.MagnifyingGlass,
+                    // We will change it to different behaviour when Search 2.5 autocomplete will be implemented
+                    query: `in:${reportForContextualSearch.reportID}`,
```

Comment:
I think there is a but here, the query that will be searched should be `in:${reportForContextualSearch.reportID}` **together** with the `currentSearch?.inputQuery`

So when I type in `test` I want the query to be: "in:123123123 test"

Reply:
You're right. I just focused on some other design doc screen shot and forgot about this, I will fix it tomorrow.